### ISSUE TITLE
[Jessie] Apache SSLMutex directive depreciated in 2.4 - Needs review

### ISSUE THREAD

**Issue by JedMeister** [turnkeylinux/tracker#351]
Despite attempts to work this out (see below) in the end, seeing as I don't really understand the implications of any of this; I think that the best course of action is to just comment it out (as per [this pull request](turnkeylinux/common/pull/14) and defer a proper fix to @lirazsiri 

**Reference material:**
[Specific note re removal of the SSLMutex directive (and others)](https://httpd.apache.org/docs/2.4/upgrading.html#misc)
[Docs regarding Mutex directive in v2.4](https://httpd.apache.org/docs/2.4/mod/core.html#mutex)

**Details:**
````
# service apache2 start
[FAIL] Starting web server: apache2 failed!
[warn] The apache2 configtest failed. ... (warning).
Output of config test was:
AH00526: Syntax error on line 50 of /etc/apache2/mods-enabled/ssl.conf:
Invalid command 'SSLMutex', perhaps misspelled or defined by a module
not included in the server configuration
Action 'configtest' failed.
The Apache error log may have more information.
````
As noted in the title, Apache2.4 (default in Jessie) the 'SSLMutex' directive has been depreciated. I read that it has been replaced with `Mutex` so I changed line 50 from:
`SSLMutex  file:${APACHE_RUN_DIR}/ssl_mutex`
to:
`Mutex  file:${APACHE_RUN_DIR}/ssl_mutex`
But then there was a new error:
````
# service apache2 start
[FAIL] Starting web server: apache2 failed!
[warn] The apache2 configtest failed. ... (warning).
Output of config test was:
AH00526: Syntax error on line 50 of /etc/apache2/mods-enabled/ssl.conf:
Invalid Mutex directory in argument file:/var/run/apache2/ssl_mutex
Action 'configtest' failed.
The Apache error log may have more information.
```
It seems that Apache has changed it's run dir from /var/run/apache2 to /run/apache2 (so the SSL Mutex dir is found at /run/apache2/ssl_mutex; not /var/run/apache2/ssl_mutex). If I make the dir (e.g. `mkdir /var/run/apache2/ssl_mutex`) then it works:
````
# service apache2 start
[....] Starting web server: apache2AH00557: apache2:
apr_sockaddr_info_get() failed for tkldev
AH00558: apache2: Could not reliably determine the server's fully
qualified domain name, using 127.0.0.1. Set the 'ServerName' directive
globally to suppress this message
. ok
```

**Comment by lirazsiri** [comment]
Rather than adapt the old configuration to the new version of Apache we should reapply the SSL configuration changes we applied in buildtasks to the new default configuration files, assuming they are even still needed in Jessie. Eventually the default configuration is doing to be secure both in terms of disabling insecure protocols such as SSLv[23] and using a secure cipher list. We just need to keep patching the configuration files until that happens.

**Comment by JedMeister** [comment]
I'm closing this one as there is lots of SSL/security updates being included in v14.0
